### ISSUE TITLE
[P2][dogfood] Python: parent function inherits a nested function's `return_type` when the parent has no return annotation

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#792]
**Severity:** P2  |  **Surface:** CLI --advanced (Python extraction, return_type field)
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
`outer` is reported with return_type='str'. `outer` has NO return annotation in source; the `str` is the nested `inner`'s `-> str` return type bleeding up to the parent. Root cause: _return_type_from_signature_text (_signature_helpers.py:94-104) and _extract_return_type_from_node (_function_extractor_mixin.py:159-168) both split the entire function node_text on the first `->`, capturing a nested/inner function's annotation.

### Expected
`outer` has no return annotation, so return_type should be null/None (or, if the project wants a default, the same `Any` default used elsewhere) — never the nested function's annotation `str`. Return-type extraction must read only the function's own signature (the function_definition node's direct `type`/return child, the text up to the signature's colon), not the first `->` found anywhere in the body. Wrong signature data misleads agents reasoning about call results.

### Repro
```bash
printf 'def outer():\n    def inner(x: int) -> str:\n        return str(x)\n    return inner\n' > /tmp/nested.py && uv run python -m tree_sitter_analyzer /tmp/nested.py --advanced
# In the output, the `outer` element has return_type="str" even though outer has no -> annotation.
```

### Verifier notes
Reproduced cleanly with the exact repro: a top-level `def outer():` with NO return annotation is reported with return_type='str', which is the nested `def inner(x: int) -> str` annotation bleeding up to the parent. Root cause confirmed in two extractor paths that split the full function node_text on the FIRST `->` anywhere in the body rather than the function's own signature: (1) tree_sitter_analyzer/languages/python_plugin/_signature_helpers.py:94-104 `_return_type_from_signature_text` (used by _parse_function_signature_optimized at _function_extractor_mixin.py:113); and (2) the mirror bug at _function_extractor_mixin.py:159-168 `_extract_return_type_from_node` (split parts[1] of full node_text). When a function has no top-level `type` child, the AST fallback (_return_type_from_type_child) cannot correct it, so the wrong text-split value survives. Not a flag/user error: --advanced is a documented mode and the field is plainly wrong. Real-world requests/status_codes.py `_init` (L109, no annotation) currently reports 'Any' in the installed version (its nested `def doc` returns its own annotation in this version), but the defect mechanism is identical and solidly demonstrated by the synthetic case — any parent with an annotated nested/sibling def reachable before its own signature inherits that type. No overlap with open issues #734-#757 or other open issues (closest #740/#742 cover is_method/receiver/class_name; #751 covers C++ nested struct extraction — different fields/languages).

**Comment by aimasteracc** [comment]
Fixed and merged via PR #831 (Python nested function return_type)
